javascript/frameworks/cap/test/queries/cqlinjection/srv/service1.js

```diff
@@ -0,0 +1,786 @@
+    });
+
+    this.on("send47", async (req) => {
+      const { id } = req.data;
+      const Service2 = await cds.connect.to("Service2");
+      Service2.delete(`Service2Entity`).where("ID =" + id);
+    });
+
+    /* ========== 5. Service1 running query on Service2 using CQN parsed with `cds.ql` ========== */
+    this.on("send51", async (req) => {
+      const { id } = req.data;
+      const Service2 = await cds.connect.to("Service2");
+      const query = cds.ql("SELECT * from Service1Entity where ID =" + id);
+      Service2.run(query);
+    });
+
+    this.on("send51", async (req) => {
+      const { id } = req.data;
+      const Service2 = await cds.connect.to("Service2");
+      const query = cds.ql(`SELECT * from Service1Entity where ID =` + id);
+      Service2.run(query);
+    });
+
+    this.on("send53", async (req) => {
+      const { id } = req.data;
+      const Service2 = await cds.connect.to("Service2");
+      const query = cds.ql(`SELECT * from Service1Entity where ID = ${id}`);
+      Service2.run(query);
+    });
+
+    this.on("send54", async (req) => {
+      const { id } = req.data;
+      const Service2 = await cds.connect.to("Service2");
+      const query = cds.ql`SELECT * from Service1Entity where ID = ${id}`;
+      Service2.run(query);
+    });
+
+    /* ========== 6. Service1 running query on the database service using CQN parsed with `cds.parse.cql` ========== */
+    this.on("send61", async (req) => {
+      const { id } = req.data;
+      const query = cds.parse.cql("SELECT * from Entity1 where ID =" + id);
+      cds.run(query);
+    });
+
+    this.on("send62", async (req) => {
+      const { id } = req.data;
+      const query = cds.parse.cql(`SELECT * from Entity1 where ID =` + id);
+      cds.run(query);
+    });
+
+    this.on("send63", async (req) => {
+      const { id } = req.data;
+      const query = cds.parse.cql(`SELECT * from Entity1 where ID = ${id}`);
+      cds.run(query);
+    });
+
+    this.on("send64", async (req) => {
+      const { id } = req.data;
+      const query = cds.parse.cql`SELECT * from Entity1 where ID = ${id}`; // FP
+      cds.run(query);
+    });
+
+    /* ========== 7. Service1 running query on the database service using CQN parsed with global function `CQL` ========== */
+    this.on("send71", async (req) => {
+      const { id } = req.data;
+      const query = CQL("SELECT * from Entity1 where ID =" + id);
+      cds.run(query);
+    });
+
+    this.on("send72", async (req) => {
+      const { id } = req.data;
+      const query = CQL(`SELECT * from Entity1 where ID =` + id);
+      cds.run(query);
+    });
+
+    this.on("send73", async (req) => {
+      const { id } = req.data;
+      const query = CQL(`SELECT * from Entity1 where ID = ${id}`);
+      cds.run(query);
+    });
+
+    this.on("send74", async (req) => {
+      const { id } = req.data;
+      const query = CQL`SELECT * from Entity1 where ID = ${id}`; // FP
```

Comment:
Is this still an FP? It doesn't appear in the expected results.

Reply:
Addressed in [28ae679](https://github.com/advanced-security/codeql-sap-js/pull/200/commits/28ae6792b81810d6cb965562be59b23a6a4612ed).